### PR TITLE
UP-4776: Support filters on body background image - master

### PR DESCRIPTION
**Pull request by ChristianMurphy**
https://issues.jasig.org/browse/UP-4776

#### Improvement

Sometimes images need to be adjusted to meet stylistic constraints or accessibility contrast requirements.
Offering filters for background image would allow images to be adjusted dynamically, without modifying the original image.
And could be used with dynamic skins to allow different tenants to re-use the same image, applying different filters to match the image with the rest of the skin.

#### Resolution

I added a `@portal-page-body-background-image-filter` variable that pairs with the existing `@portal-page-body-background-image` to allow filtering.

#### Example

*Standard*

screencapture-localhost-8080-uportal-f-welcome-normal-render-up-1480446079342

*Darkened*

screencapture-localhost-8080-uportal-f-welcome-normal-render-up-1480446790245

*Sepia*

screencapture-localhost-8080-uportal-f-welcome-normal-render-up-1480447233358

### CODE REVIEW

**Comment by bjagg** [comment]
@ChristianMurphy Would you mind adding a Markdown doc in the docs/ directory with these sample images?

**Comment by ChristianMurphy** [comment]
@bjagg Can do.
What would you be looking for beside the images in the document?
A markdown version of the [uPortal Skinning Guide](https://wiki.jasig.org/display/UPM42/Skinning+and+Branding) perhaps?

**Comment by ChristianMurphy** [comment]
@bjagg I added a skinning guide adapted from the [4.3 docs](https://wiki.jasig.org/display/UPM43/Skinning+and+Branding) and the [4.2 docs](https://wiki.jasig.org/display/UPM42/Skinning+and+Branding).
I'm working on adding some documentation about the custom uPortal style variables now.

File: uportal-war/src/main/webapp/media/skins/respondr/defaultSkin.less [hunk 1]
```diff
@@ -30,10 +30,9 @@
  * See https://wiki.jasig.org/display/UPC/Respondr+Skinning+Process for instructions on creating a custom skin.
  */
 
-/** DO NOT REMOVE THIS INCLUDE ****************/
-@import "defaultSkin/less/variables.less"; 
-/**********************************************/
-
+/***** DO NOT REMOVE THESE INCLUDES *****/
```
Comment (drewwills) [inline uportal-war/src/main/webapp/media/skins/respondr/defaultSkin.less]: How about 'DO NOT REMOVE THESE INCLUDES OR ALTER THEM IN ANY WAY'

File: uportal-war/src/main/webapp/media/skins/respondr/defaultSkin.less [hunk 2]
```diff
@@ -30,10 +30,9 @@
...
+@import "defaultSkin/less/variables.less";
 @import "common/common.less";
+/****************************************/
```
Comment (drewwills) [inline uportal-war/src/main/webapp/media/skins/respondr/defaultSkin.less]: Beneath this line, maybe we should add...

/**
 * To  create your own skin based on defaultSkin, uncomment the following line
 * and replace {skinName} with the name of your skin.
 */
/* @import "{skinName}/less/variables.less"; */

Comment (ChristianMurphy) [inline uportal-war/src/main/webapp/media/skins/respondr/defaultSkin.less]: Maybe add a reference to the markdown document in the less file? e.g.

``` less
// For instructions on skinning uPortal
// @see docs/SKINNING_UPORTAL.md
```

**Comment by cousquer** [comment]
👍 @ChristianMurphy, @drewwills,
Many Thanks for raising the concept of non breaking change ( [semver minor](http://semver.org/) )

**Comment by ChristianMurphy** [comment]
Added a fix to allow customize region to work with background filters

screencapture-localhost-8080-uportal-f-admin-normal-render-up-1481316642238

screencapture-localhost-8080-uportal-f-admin-normal-render-up-1481316651354

**Comment by ChristianMurphy** [comment]
Copied over the skinning documentation and added a section on special notes.
https://github.com/ChristianMurphy/uPortal/blob/fecdb96eb80835c07006737bc08d30c57d429229/docs/SKINNING_UPORTAL.md

**Comment by ChristianMurphy** [comment]
@bjagg @drewwills requested changes have been added :white_check_mark: , could you take another look? :eyes:

File: uportal-war/src/main/webapp/media/skins/respondr/defaultSkin.less [hunk 3]
```diff
@@ -16,24 +16,15 @@
...
 
 /**
- * Import order:
- * 1. defaultSkin/less/variables.less
- * 2. common/less/common.less  - see comments in common.less file for more information
- * 3. defaultSkin/less/skin.less
- *
- * See https://wiki.jasig.org/display/UPC/Respondr+Skinning+Process for instructions on creating a custom skin.
+ * For information on how to skin uPortal
+ * @see docs/SKINNING_UPORTAL.md
```
Comment (bjagg) [inline uportal-war/src/main/webapp/media/skins/respondr/defaultSkin.less]: (+1)

**Pull request merged**